cli/utils/promisfied-fs.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * code based on appcenter-cli
+ */
+
+const { promises: fs } = require('fs');
+const path = require('path');
+
+/**
+ *
+ * @param dir {string}
+ * @return {Promise<string[]>}
+ */
+async function walk(dir) {
+    const stats = await fs.stat(dir);
+    if (stats.isDirectory()) {
+        /**
+         * @type {string[]}
+         */
+        let files = [];
+        for (const file of await fs.readdir(dir)) {
+            files = files.concat(await walk(path.join(dir, file)));
```

Comment:
As far as I know, `path.join`'s type is `join(...paths: string[]): string;`.

```suggestion
            files = files.concat(await walk(path.join(...dir, file)));
```